### ISSUE TITLE
Broken in higher-order components

### ISSUE THREAD

**Issue by ctumolosus** [layershifter/babel-plugin-transform-react-handled-props#8]
I tried using this plugin in my project but got the following error when compiling higher-order components:

```
We don't know what to do with this node type. We were previously a Statement but we can't fit in here?
    at NodePath.insertAfter (~/babel-traverse/lib/path/modification.js:175:13)
    at insertAfterPath (~/node_modules/babel-plugin-transform-react-handled-props/lib/util/appendProps.js:50:27)
    at ~/node_modules/babel-plugin-transform-react-handled-props/lib/util/appendProps.js:59:5
```

Here is a snippet reproducing the problem:

```
import React from 'react';

export default function WithSauce(options = {}) {
  return function createContainer(Component) {
    return class Sauce extends React.Component {
      render() {
        return (
          <Component className="sauced" />
        );
      }
    };
  };
}
```

I tried changing my syntax a bit and got a different problem (possibly unrelated)

```
export default (options) => (Component) => class WithSauce extends React.Component { [...] }
```

Transpiled to:

```
module.exports = function (options) {
  return function (Component) {
    return function WithSauce() { [...] }
  }
}

WithSauce.handledProps = []; // Error: WithSauce is not defined.
```

Let me know if there is anything else I can help with to identify the root cause.

**Comment by layershifter** [comment]
@ctumolosus Thanks for detailed report. I will fix it as soon as i can.

**Comment by gadyonysh** [comment]
How about fix?

**Comment by layershifter** [comment]
I will fix this issue on this week. Thanks for notify

**Comment by layershifter** [comment]
Sorry for delay, guys.
Fixed in `0.2.4`. Will be nice to have a feedback there.

**Comment by gadyonysh** [comment]
Works fine for me. Thank you for fixing.
